drivers/docker/config_test.go

```diff
@@ -41,6 +42,32 @@ func TestConfig_ParseHCL(t *testing.T) {
 	}
 }
 
+func TestConfig_ParseJSON_Regression(t *testing.T) {
+	cases := []struct {
+		name  string
+		input map[string]interface{}
+	}{
+		{
+			name:  "nil values for blocks are safe",
+			input: map[string]interface{}{"image": "bash:3", "mounts": nil},
+		},
+	}
+
+	spec, diag := hclspecutils.Convert(taskConfigSpec)
+	require.Empty(t, diag)
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.name, func(t *testing.T) {
+			// TODO(dani): Expand these tests to validate the produced TaskConfig
+			//             when bouncing through messagepack.
+			v, diag := hclutils.ParseHclInterface(c.input, spec, nil)
+			require.Empty(t, diag)
+			require.NotNil(t, v)
```

Comment:
You can use the helper function for msgpack stuff like, like how `TestConfig_ParseHCL` test does but it uses json text input though:

```
cfgStr := `{"config": {"image": "bash:4", "mounts": nil}}`
var tc *TaskConfig
hclutils.NewConfigParser(taskConfigSpec).ParseJson(t, cfgStr, &tc)
```

Reply:
Sweet I had trouble with manually handling messagepack for some non obvious reasons - updated to use the helper